retry-backoff/retry-backoff.ts

```diff
@@ -0,0 +1,87 @@
+import {
+  createContext,
+  type Context as ContextType,
+  type Operation,
+  race,
+  sleep,
+} from "npm:effection@4.0.0-alpha.3";
+import prettyMilliseconds from "npm:pretty-ms@7.0.1";
+
+interface UseRetryBackoffOptions {
+  timeout?: number;
+}
+
+export interface RetryWithContextDefaults {
+  timeout: number;
+}
+
+const RetryWithBackoffContext = createContext<RetryWithContextDefaults>(
+  "retry-with-context",
+  {
+    timeout: 30_000,
+  }
+);
+
+export function* useRetryWithBackoff<T> (
```

Comment:
We need docs for this function.

Reply:
That's right. You don't need to describe the parameters and properties because those will be generated from typescript